### PR TITLE
Teams: allow split stride of zero iff size is 1

### PR DESCRIPTION
**Pull request by davidozog**
This PR allows team split operations to have a stride of zero if and only if the new team size is 1.

### CODE REVIEW

File: src/shmem_team.c [hunk 1]
```diff
@@ -303,7 +303,7 @@ int shmem_internal_team_split_strided(shmem_internal_team_t *parent_team, int PE
 
     if (PE_start < 0 || PE_start >= parent_team->size ||
         PE_size <= 0 || PE_size > parent_team->size   ||
-        PE_stride == 0) {
+        (PE_stride == 0 && PE_size !=1)) {
```
[the remark this answers is not in the document]
Comment (davidozog) [reply src/shmem_team.c:306]: oops!  Good eye! 👀

[the remark this answers is not in the document]
Comment (davidozog) [reply src/shmem_team.c]: I amended the commit in https://github.com/Sandia-OpenSHMEM/SOS/pull/1136/commits/28b6361c7ab54cd71c04945138796758c043dcb6

[the remark this answers is not in the document]
Comment (wrrobin) [reply src/shmem_team.c]: Looks like we allow any `PE_stride` when `PE_size` equals 1. In such case, should we assign `myteam->stride` to 0 or the supplied value?

**Review by wrrobin — APPROVED** [review]

**Comment by davidozog** [comment]
@wrrobin @philipmarshall21 - I reverted the zero-stride checks across collectives (https://github.com/Sandia-OpenSHMEM/SOS/pull/1136/commits/998f7c9b5d9159a20efd24418438af1704e83577) by forcing `stride=1` internally, which I think is a little cleaner.  Can you please re-review?

**Review by wrrobin — APPROVED** [review]
LGTM. So, internally, we are not allowing stride to be 0. Perhaps, we should discuss if we need to revisit this in OpenSHMEM.

**Comment by davidozog** [comment]
> LGTM. So, internally, we are not allowing stride to be 0. Perhaps, we should discuss if we need to revisit this in OpenSHMEM.

Yes, agreed.  Fortunately I think the spec is arguably well-defined if the following is true:  `shmem_team_split_strided` allows `stride` to be zero only if `size` is 1 (but `stride` can also be _any_ value when `size` is 1).  So it's up to library implementers to handle the special zero-stride case, because it's likely to lead to divide-by-zero exceptions.

I've also drafted a change here that should help clarify this in the spec:
https://github.com/davidozog/openshmem-specification/commit/010e42489d89e8088dea84c447d283e04651afd3

I will propose it for v1.6 if there's enough time...

**Review by philipmarshall21 — APPROVED** [review]
LGTM!

**Pull request merged**